ftw/globalstatusmessage/tests/test_controlpanel.py

```diff
@@ -153,3 +157,32 @@ def test_receiver(self, browser):
             },
             self.expected_settings
         )
+
+    @browsing
+    def test_receiver_filters_not_exitsing_paths_before_setting_theme(self, browser):
+        subsite = create(Builder('folder')
+                         .titled(u'subsite')
+                         .providing(INavigationRoot))
+        subsite_path = '/'.join(subsite.getPhysicalPath())
+
+        settings = deepcopy(self.expected_settings)
+        settings['exclude_sites'] = [subsite_path, '/not/existing']
+        payload = json.dumps(decode_for_json(settings))
+
+        browser.login().open(
+            self.portal,
+            view='@@global_statusmessage_config_receiver',
+            data={'jsondata': payload},
+            send_authenticator=True)
+
+        registry = getUtility(IRegistry)
+        store = registry.forInterface(IStatusMessageConfigForm, check=False)
+
+        settings['exclude_sites'].remove('/not/existing')
+        self.assertDictEqual(
+            {
+                field_name: getattr(store, field_name)
+                for field_name in getFieldNames(IStatusMessageConfigForm)
+            },
```

Comment:
I tend to assert hard-coded/non-computed values because it is hard for me to evaluate the result of this dict comprehension in my head while reading the code of this test.

Reply:
Me too. But this was already there in another test, so I kept it for local consistence.

Reply:
I see...